File: PRACTICA#2/COMPOSICION-AGREGACION/3/JAVA/Avion.java
import java.util.ArrayList;
import java.util.List;

public class Avion {
    private String modelo;
    private String fabricante;
    private List<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    //A
    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public List<Parte> getPartes() {
        return partes;
    }

    public void setPartes(List<Parte> partes) {
        this.partes = partes;
    }

    public void agregar_parte(Parte parte) {
        partes.add(parte);
    }
    public void mostrar_avión() {
        System.out.println("Modelo: " + modelo + ", Fabricante: " + fabricante);
        System.out.println("Partes:");
        for (Parte parte : partes) {
            parte.mostrar_info();
        }
    }
}

File: PRACTICA#2/COMPOSICION-AGREGACION/3/PYTHON/Avion.py
class Avion:
    def __init__(self, modelo, fabricante):
        self.__modelo = modelo
        self.__fabricante = fabricante
        self.__partes = []

    #A
    def get_modelo(self):
        return self.__modelo

    def set_modelo(self, modelo):
        self.__modelo = modelo

    def get_fabricante(self):
        return self.__fabricante

    def set_fabricante(self, fabricante):
        self.__fabricante = fabricante

    def get_partes(self):
        return self.__partes

    def set_partes(self, partes):
        self.__partes = partes

    def agregar_parte(self, parte):
        self.__partes.append(parte)

    def mostrar_avión(self):
        print(f"Modelo: {self.__modelo}, Fabricante: {self.__fabricante}")
        print("Partes:")
        for parte in self.__partes:
            parte.mostrar_info()